### ISSUE TITLE
Lint about range intersections in `if … else if` sequences

### ISSUE THREAD

**Issue by mcarton** [Manishearth/rust-clippy#1180]
We could warn for stuff like that:

```rust
if (a < 42) {
    …
} else (a < 7) { // It is commonly admitted that 7 < 42
    …
}
```

See http://www.viva64.com/en/examples/V695/. Cc #1175.

**Comment by oli-obk** [comment]
This and many others should be part of rustc's dead code lint imo.
